### PR TITLE
build: all required libraries by libegg should be in LIBEGG_LIBS

### PR DESCRIPTION
**Pull request by rbuj**
See https://github.com/mate-desktop/engrampa/pull/394

### CODE REVIEW

**Comment by raveit65** [comment]
Just rebase to get lastest travis commit in PR.

**Comment by rbuj** [comment]
Now we have a configure.ac ...

**Comment by raveit65** [comment]
> Now we have a configure.ac ...

Doesn't matter it is only used for travis, They won't get include in tarballs of packages.
Edit:
And build time is very fast now :)

File: libegg/Makefile.am [hunk 1]
```diff
@@ -32,6 +32,6 @@ libegg_la_CFLAGS =				\
 
 libegg_la_LIBADD = 	\
 	$(LIBEGG_LIBS)	\
-	-lSM -lICE
```
Comment (raveit65) [inline libegg/Makefile.am:35]: Maybe this is missing when we build mate-submodule in a host package, or not?
Rest looks OK.

**Review by yetist — APPROVED** [review]
LGTM

File: libegg/Makefile.am (hunk 1 above)
Comment (raveit65) [reply libegg/Makefile.am:35]: no answer...
we can fix it later, if it is a problem.

**Comment by raveit65** [comment]
Branch is restored. So this can be improved.
I will revert this commit on master to fix caja building for the moment.

File: libegg/Makefile.am (hunk 1 above)
Comment (rbuj) [reply libegg/Makefile.am:35]: This submodule was checked with the build of Engrampa, for this reason I added a link to the change necessary to Engrampa to use this change on the PR comment body.

While the merge request was active, Engrampa was dispensed and the configure.ac file was added to test the isolated build of the submodule. I added the required change to configure.ac to use this change.

Yes, to use this change you need to update the projects that use this submodule.

**Comment by raveit65** [comment]
Anything left what you need from this PR?

**Comment by rbuj** [comment]
This change is still necessary in order to do not include `-lSM -lICE` twice:
```
$ grep "^LIBEGG_LIBS =.*" libegg/Makefile
LIBEGG_LIBS = -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lSM -lICE 
```
change:
```patch
diff --git a/libegg/Makefile.am b/libegg/Makefile.am
index 3601740..7bcdb72 100644
--- a/libegg/Makefile.am
+++ b/libegg/Makefile.am
@@ -32,6 +32,6 @@ libegg_la_CFLAGS =                            \
 
 libegg_la_LIBADD =     \
        $(LIBEGG_LIBS)  \
-       -lSM -lICE
+       $(NULL)
 
 -include $(top_srcdir)/git.mk
```
Test (... -lSM -lICE  -lSM -lICE)
```
$ autoreconf -fi && ./configure && make V=1 &> make.log
$ fgrep "ICE" make.log 
/bin/sh ../libtool  --tag=CC   --mode=link gcc -DEGG_SM_CLIENT_BACKEND_XSMP -DG_LOG_DOMAIN=\""EggSMClient"\" -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/fribidi -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/atk-1.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/at-spi-2.0 -pthread  -Wall -Wmissing-prototypes     -o libegg.la  libegg_la-eggtreemultidnd.lo  libegg_la-eggdesktopfile.lo libegg_la-eggsmclient.lo libegg_la-eggsmclient-xsmp.lo   -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lSM -lICE  -lSM -lICE 
```

**Comment by raveit65** [comment]
ok, go ahead.
Just commit directly.

**Pull request merged**